common-util/Details/ServiceState/3rdStepFinishedRegistration/utils.jsx

```diff
@@ -85,39 +94,131 @@ export const handleMultisigSubmit = async ({
 
   // signer
   const provider = new ethers.providers.Web3Provider(
-    window.web3.currentProvider,
+    window.MODAL_PROVIDER || window.web3.currentProvider,
     'any',
   );
-  await provider.send('eth_requestAccounts', []);
-  const signer = provider.getSigner();
-
-  // Get the signature of a multisend transaction
-  const signatureBytes = await signer._signTypedData(
-    { verifyingContract: multisig, chainId },
-    EIP712_SAFE_TX_TYPE,
-    safeTx,
-  );
+  const code = await provider.getCode(account);
 
-  const safeExecData = multisigContract.interface.encodeFunctionData(
-    'execTransaction',
-    [
-      safeTx.to,
-      safeTx.value,
-      safeTx.data,
-      safeTx.operation,
-      safeTx.safeTxGas,
-      safeTx.baseGas,
-      safeTx.gasPrice,
-      safeTx.gasToken,
-      safeTx.refundReceiver,
-      signatureBytes,
-    ],
-  );
+  try {
+    // TODO: check if we are dealing with safe in future!
+    // gnosis-safe
+    if (code !== '0x') {
+      // Create a message data from the multisend transaction
+      const messageHash = await multisigContract.getTransactionHash(
+        safeTx.to,
+        safeTx.value,
+        safeTx.data,
+        safeTx.operation,
+        safeTx.safeTxGas,
+        safeTx.baseGas,
+        safeTx.gasPrice,
+        safeTx.gasToken,
+        safeTx.refundReceiver,
+        nonce,
+      );
 
-  const packedData = ethers.utils.solidityPack(
-    ['address', 'bytes'],
-    [multisig, safeExecData],
-  );
+      const multisigContractServiceOwner = getServiceOwnerMultisigContract(multisig);
+      const startingBlock = await provider.getBlockNumber();
+
+      // Get the signature bytes based on the account address, since it had its tx pre-approved
+      const signatureBytes = `0x000000000000000000000000${account.slice(
+        2,
+      )}${ZEROS}01`;
```

Comment:
@mohandast52 have you checked that you are getting the same value here as before after this optimization?

Also you could use '0'.repeat(first_number_of_zeros), then account.slice, then '0'.repeat(second_number_of_zeros), then '01'